runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleExtensions.kt

```diff
@@ -61,3 +61,42 @@ public infix fun SerialModule.overwriteWith(other: SerialModule): SerialModule =
         }
     })
 }
+
+/**
+ * Looks up a descriptor of serializer registered for contextual serialization in [this],
+ * using [SerialDescriptor.capturedKClass] as a key.
+ *
+ * @see SerialModule.getContextual
+ * @see SerializersModuleBuilder.contextual
+ */
+public fun SerialModule.getContextualDescriptor(descriptor: SerialDescriptor): SerialDescriptor? =
+    descriptor.capturedKClass?.let { klass -> getContextual(klass)?.descriptor }
+
+/**
+ * Retrieves a collection of descriptors which serializers are registered for polymorphic serialization in [this]
+ * with base class equal to [descriptor]'s [SerialDescriptor.capturedKClass].
+ *
+ * @see SerialModule.getPolymorphic
+ * @see SerializersModuleBuilder.polymorphic
+ */
+public fun SerialModule.getPolymorphicDescriptors(descriptor: SerialDescriptor): List<SerialDescriptor> {
+    val kClass = descriptor.capturedKClass ?: return emptyList()
+    // shortcut
+    if (this is SerialModuleImpl) return this.polyBase2Serializers[kClass]?.values.orEmpty()
+        .map { it.descriptor }
+
+    val builder = ArrayList<SerialDescriptor>()
+    dumpTo(object : SerialModuleCollector {
+        override fun <T : Any> contextual(kClass: KClass<T>, serializer: KSerializer<T>) { /*noop*/
+        }
+
+        override fun <Base : Any, Sub : Base> polymorphic(
+            baseClass: KClass<Base>,
+            actualClass: KClass<Sub>,
+            actualSerializer: KSerializer<Sub>
+        ) {
+            if (baseClass == kClass) builder.add(actualSerializer.descriptor)
+        }
+    })
+    return builder
+}
```

Comment:
When reading this code I believe this is almost the missing piece I've been waiting for. For my XML format I must be able to enumerate all possible children of a parent (so I can determine the tag names to recognize and which property they belong to). I need the same information for descriptors of lists (IOW. I need to get the list element descriptor -- btw. I haven't checked that this information isn't available).